packages/components/src/input-control/reducer/reducer.ts

```diff
@@ -224,10 +226,10 @@ export function useInputControlStateReducer(
 			initialState.value !== currentState.current.value &&
 			! currentState.current.isDirty
 		) {
-			reset(
-				initialState.value,
-				currentState.current._event as SyntheticEvent
-			);
+			dispatch( {
+				type: actions.RESET,
+				payload: { value: initialState.value },
+			} );
```

Comment:
Two reasons for switching to `dispatch`:
- `reset` is typed to require passing an event for the second argument and I didn't want to change the typing because in most cases it would be important to pass the event.
- The `exhaustive-deps` lint rule knows `dispatch` is stable and won't complain when it's not in the dependencies.

My remaining peeve with this is I think it'd be better to not reuse `RESET` here and maybe reintroduce `UPDATE`. It's only a theoretical concern only and for now a non-issue. The concern would be that someday a component wants to specialize `RESET` in a way that should not happen every time the value prop has changed.

Reply:
Thanks for the explanation of `dispatch` over `reset`, it makes sense. I also agree regarding `RESET` vs `UPDATE` and that it is a potential future concern but that can be addressed down the line and separately to this PR.

Reply:
This sounds like a good point, if we were to keep the internal reducer!

At the same time, after this fix is merged I'd like to explore ways to simplify `InputControl` (and consequently its derived components), including the removal of the internal reducer (see [this comment](https://github.com/WordPress/gutenberg/pull/41264#issuecomment-1139077885) for more details)